geode-core/src/main/java/org/apache/geode/internal/cache/PRHARedundancyProvider.java

```diff
@@ -1707,21 +1707,12 @@ public boolean recoverPersistentBuckets() {
     ArrayList<ProxyBucketRegion> bucketsHostedLocally =
         new ArrayList<ProxyBucketRegion>(proxyBucketArray.length);
 
+    allBucketsRecoveredFromDisk = new CountDownLatch(proxyBucketArray.length);
```

Comment:
how about moving this down to AFTER the RedundancyLogger is created.
That way if the logger creation fails it will not even have created the CountDownLatck